test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/InteroperabilityTest.scala

```diff
@@ -630,13 +630,15 @@ class InteroperabilityTest {
   @Test def boxValueClassesGivenToJSInteropMethod(): Unit = {
     val obj = js.eval("""
       var obj = {
-        stringOf: function(vc) { return vc.toString(); }
+        test: function(vc) { return vc; }
       };
       obj;
     """).asInstanceOf[InteroperabilityTestValueClassParam]
 
     val vc = new SomeValueClass(7)
-    assertEquals("SomeValueClass(7)", obj.stringOf(vc))
+    val r = obj.test(vc)
+    assertTrue(r.isInstanceOf[SomeValueClass])
+    assertEquals(7, r.asInstanceOf[SomeValueClass].i)
```

Comment:
IMO this weakens the test: Now we only test that round-tripping a value class via JS leaves it boxed.

IIUC on WASM this is the best we can do. Have you considered adding/keeping a more specific test for non-WASM?

Reply:
Good point. I added a test to `ExportsTest` to compensate (and even test a bit more).